### ISSUE TITLE
AgileVentures - thoughts on moving forward

### ISSUE THREAD

**Issue by freeranger** [AgileVentures/WebsiteOne#745]
**Developers**

I think for new developers, there are a few barriers to entry:-

* I don't know what skills the projects need so I don't know if I could help. If you look at http://up-for-grabs.net/#/tags/ you can zero in on the projects that match your skillset so you find ones more "relevant" to you.
* My experience with the technology is a bit shakey so I'm afraid I will commit to something beyond my skillset.  Some projects tag issues with an indicaton of how easy/difficult they are - e.g. https://github.com/influxdata/influxdb/labels/status%2Fhelp-wanted - so maybe I start with some easy/quick issues to get my feet wet, get some confidence, and move on from there.
* I'm scared of breaking things - I don't know the project, maybe I'm new to this whole contributing to open source game. Perhaps an experienced team member could guide me through my first issue or two? Maybe there is a sample project for newbies to play around with and contribute to and see how it all works, somewhere they don't need to be afraid of contributing.
* I can't commit to pairing - My availability is all over the place, I'm not sure I can commit to meeting up online with someone at certain times, or my connection is really bad or many other reasons I can't pair, or I just don't like the idea of it. Maybe I know the benefits of pairing and I would encourage people to try it but maybe it is emphasised too much and I want to start off without pairing, submtting pull requests for someone to review. Basically not wanting or not being able to pair shouldn't be a barrier to contributing to a project.
* Limited technologies. There seems to be a clear "bias" towards Ruby here, which is fair enough, but what if I am an awesome .NET developer or node.js developer and would like to work on a project using those skills - is this possible? Is AV "meant" to be a ruby shop or is that just the way it has panned out?
* I can't easily find projects I'm interested in - I want to find projects particularly about saving badgers, or the most active projects or the newest projects, or the "hottest" projects (most ractivity recently? Most developers joinging recently?) - there's no way to do that, I just have to trawl the pages.

**Charities**

You need to attract new business from charities in order to continue to grow, so you need to make charities want to use you, which means it must be easy for them to find you and find what you do and be confident you can meet their needs.

*Scenario* - I am a charity looking to have a site made, I google for <whatever term turns up agileventures as an option> and go to agileventures.org

The first thing I see is stuff about pairing and learning to code and developing my team skills. But I am not a developer, I am a charity - this site seems focused on coders, not on businesses looking for a site to be developed. It doesn't seem relevant to me at all, so I go elsewhere...

Lets say I get past the front screen because maybe you are relevant after all. I want to see what you guys have been doing to gauge if you are any good or not. Where do I go? hmm, I click into "Projects" as the most likely place to get a feel for what you do and...I get a random list of projects.

I really want to see one of the actual sites in action, so I pick one at random, say LocalSupport, (Note, I tried a few other at random such as MetPlus, but I couldn't a link to the site at all in the ones I tried!) in the hope that I will get a link to click on to the actual site. Hmm..nothing obvious - though I notice under the title of the site, the page says "VAH's Local Support site is at www.harrowcn.org.uk" but this isn't even a clickable link, so I have to copy and paste it....oh wait, further down the page there is an actual clickable link to the LocalSupport site, so I can finally see one of the sites you have developed.

Boy that was hard work! Other projects don't even seem to have any links - I have no idea (as a developer or as a potential customer) if these sites are live at all, if they are just at the early alpha stage, if they have even gotten off the ground.

What I would like to see is a very obvious link, front and centre, to the current production site. Regardless of whether the site is live or not, I would also like a link to the latest deployment of the development/UAT instance so I can see what it looks like - for not-yet-live sites I can now see what they are about and for currently live sites, I can see what's due to be released soon presumably.

It would be nice to see a thumbnail of the site on the project page also.

I know you said in the meeting that you didn't think a showcase was that important at the moment, but if you want to attract new customers then you need to make it more customer-friendly and a showcase is one way to do that - or, for now, just spruce up the project pages and make them more "discoverable" and easy to see what they look like.

**Premium Services**

There are two potential sources of revenue here - the charities and the developers:

_Charities_

Free service gets people working on your site as and when, so don't hold your breath for it to be developed...

Paid for at different levels can offer things like:

Your project is promoted on the site so hopefully will attract more developers
We could offer to pay the developers a small amount, thus likely attracting more people to it. Not sure how you would work out how to pay them - per commit? per lines of code? per feature? If stories were assigned points then a fee per story point? So a 1 point story gets you X, a 2 point story gets you 2X etc?
We could specifically direct developers to your project
We could promote your charity (as opposed to the project itself) since after all the website is a means to an end - the charity itself is the main thing.

_Developers_

This is a bit tricky since the developers are already giving their free time to help out on projects and now you want some money from them too? Hmmm...

The "fee" would have to be quite small methinks because of said donation of time to projects, but the sorts of things you could offer are:

They get first dibs on any paid for work - the premium members are made aware of it for a period of time before it is opened out to everyone,
They get some kind of ranking on your site in the members directory
Other members can "endorse" their skills (like on LinkedIn)
Maybe people who have worked with them will write testimonials about them on their member page (can help them get other paid work)
I think the area to concentrate on here would be helping the developers to promote themselves in order to get paid for work elsewhere, since we all need to make a living right?

**Comment by tansaku** [comment]
@freeranger thanks for this input - really appreciate it.   Some thoughts:

Developers
----------

* that **up-for-grabs site** is awesome and we should get ourselves involved or copy the format.  As I mentioned in slack, one issue we have is that most active projects are not currently using the same ticket management system that up-for-grabs uses
* **indication of difficulty labels** - yes - we should have more of this.  We do currently have a points system of 1,2,3 etc., in many projects where 1 represents a simple feature and 3 represents a more complex one, but perhaps we need to get that across better
* **sample project**, **experienced team guides** - yes, both great ideas, although I'm wary of sample projects as part of the idea here is to learn with real projects.  Also you *cannot* break things with the pull request model we are using - happy to explain further.  Also the whole point of the online standups is so that new members can meet other team members and then be guided.  That's how we've onboarded lots of people, but maybe we are not properly communicating that that's available, or maybe people are just too shy ...? :-)
* **pairing**, yes, I think I start to agree that we need to better support solo contributions as a gateway to pairing.  What I worry is with people working on things by themselves and then having their contributions not pulled in because they haven't been in good communication with the team upfront, but perhaps that's just the way it has to be
* We have a **Ruby/Rails bias** because the group who started the site were alumni of the 169 Ruby/Rails course.  We do have some other projects in other languages, but none of them have become hugely active.  Other languages include Python, Go and JavaScript.  Anyone can start a project in any language ...
* finding interesting projects - we do have a **search** mechanism, but perhaps it is not visible enough -- maybe we need more obvious filters?

Charities
---------

I totally agree that the current site is totally not right for attracting charities, and ultimately we do need to attract them. We tried building a separate [site](https://evening-oasis-1495.herokuapp.com/) as a prototyping exercise with a group of novice developers, with a separate site being easier for them to work with, but the result wasn't something we could immediately use.

When I say that the showcase is less important right now, it's because I feel like we need to improve things for onboarding developers before we start reaching out to more charities, so that when charities do come to us, we'll have developers that can work with them.  Also, from where I'm sitting we have lots of work to do right now, and I worry that if we focused immediately of bringing in lots of charities we'd be turning them away as there was no one to work on their projects.

Of course it's all a bit chicken and egg, but right now the site is costing me a certain amount of money to run each month, and I need to put something in place to offset that burn.  By registering as an official charity we'll get persistent slack history and be all above board for accepting donations and charging for membership of the charity.  I have various people who are telling me they'd like to supply funds, so if we can get the payment infrastructure set up for that, then I can at least stop burning cash each month :-)

However with that in place, and some basic UI improvements to the main site, perhaps then the focus really should be charities.

Premium Service
------------

Agreed, there are at least two potential sources of revenue and others as well (hiring fees from companies like freecodecamp).  6 months ago I was thinking of charities paying a listing fee; however we've yet to validate that as a business model.  I think we would have to have a showcase in place for that to get off the ground.

However I do have developers saying they'd like to contribute to the cause, so that seems like the stronger lead right now in terms of making the current site sustainable.      Making the site a nice place for developers to hang out and giving them some desirable benefits from a premium service feels like the highest priority to me at the moment; but I'm totally open to change.

Footnote
----------

BTW, I totally take your point about the problems navigating the site.  I think they affect both developers and charities, and the site is the way it is because it's evolved organically that way, rather than intentionally being set up that way.  This is largely because it was a "sample project" where novice developers and designers were practising their skills.

That was fine at the time, but my sense is that now, for the future success of the project, we need to have the site adhere to more of a central vision that provides consistency of user experience; and allows our key stakeholders to find what they need.

**Comment by freeranger** [comment]
@tansaku: Sure, glad to be able to give some input on this.
# Developers #
- **up-for-grabs.net** - do we need the whole thing, or just a way to query and link to the github repo's for the actual projects?  You mentioned elsewhere having a version of this for Pivotal...but is that a "must have"?  I guess it depends on if you want to track all issues through pivotal or through the git repo's.
To my mind, pivotal is good for running sprints, but if it's just a bunch of dev's contributing as and when, then maybe it doesn't work so well and the normal git issue mechanism is fine?
- **difficulty** - Ok yeah I found that once I went into a story in pivotal...I was looking in github.  Now I'm confused (as joe developer) as to where I look for issues - github which I know well and can see some issues on, or this other thing pivotal which I know nothing about?  Hmm there are issues in both places - does one supercede another?  Do they complement each other? I don't know.  
Also, as a newbie developer, or new to agile developer, I don't really get the concept of story points - tell me something is easy/medium/hard/a few minutes work and I get that, but what does it mean when I look in this pivotal tracker thing and a story has "points"?  What is a "point"?  All I want to know is if this is likely to suite my n00b skills because basically a monkey could do it...
I think you might need to consider if you want to be sprint oriented or if it's a more "traditional" opensource free-for-all with random contributions any time, from anyone...
Also all being on the same tracking system would be good for consistency and familiarity to people hopping between projects.  Since they are all under the Agile Ventures "umbrella", there should be some consistency to the experience across projects.

- **sample project** The idea here was for learning how AV works, how to submit to a project and get it released etc, not for learning how to be a developer.  I've got my new, shaky ruby skills and I'm willing to give it a go but I just want to see how it hangs together without affecting anything important.  
This could be something as "simple" as a tutorial showing users how to raise an issue, take ownership of an issue, make a code change, create a pull request, accept a pull request, get the software into the wild.  
Maybe they just add their name to a list displayed onscreen or something like that - a simple, visible result of getting an issue all the way  through from start to finish, and having to wear different hats in the process since you wouldn't normally raise the issue yourself and do everything else along the way too.
What this isn't is a sample project for learning to code or add any real features to anything

- **guidance** - Perhaps there is an element of shyness or embarrassment that you are a n00b and are intimidated by the online meeting idea, or perhaps you don't have video capability, or indeed perhaps you just can't commit to a specific time every week.  You guys have settled on Friday's at 5 for the AV standups it seems but that is definitely not a time that would ever suit me, and maybe a time that does suit me doesn't suit everyone else...it's hard to please everyone.  Does this mean I can't contribute?  Well clearly not since we are having this dialog, but I might feel I can't contribute if I can't (or won't) attend the standups...so perhaps you need to emphasise more that they are there to get to know each other and to help etc, but they are not mandatory for participation in projects?  
Ditto pair programming...unless of course you *want* it to be mandatory... You would need to make it clear though that if you aren't pairing "live" then you do need to have a dialog going with at least one other person and effectively pair (or at least code review) "asynchronously" as it were.  This way your pull requests don't get left by the wayside and you don't have to worry that you have done something stupid on your own.
For this you may need people to volunteer to be mentors to n00bies to help them through the early days of being an AV contributor.

- **languages** - I can understand why it started out as ruby-centric....looking around, it would still seem to be that way and there is nothing to suggest other languages are up for grabs (C# for example? java? scala?) so I might wrongly assume it is exclusively for ruby-ists.  There is nothing in the "Getting started" section for example that suggests I can use any technology I like.  Speaking of which...

- **getting started** - The "barrier to entry" seems quite high.  Step 1 wants me to:
_Setup accounts for GitHub, Pivotal Tracker and Google+. Also, make sure that you have a YouTube Channel connected to your G+ account.  These services are crucial for you to be able to participate in AgileVentures daily activities ._ - wow, that's a lot to even get started - think I'll take my project to github instead, where I can get going in seconds...
- **search** Found it!  But I was looking on the projects page for a search feature for projects - this is a hidden-in-plain-sight search across the website.  I am trying to zero in on projects by technology perhaps or the sorts of charity I am interested in.  My sister had breast cancer, so maybe I want to contribute to breast cancer related sites - how do I discover those? 
Or I came here looking for C# projects but on the "look for projects you are interested in" page I see one that is using Angular...hmmm I'm interested in that too - maybe I will check that out instead.  Discoverable see?  up-for-grabs lists "popular" tags front and centre on the projects page - that's a good idea...

## Charities ##
Chicken and egg as you say - the charities presumably would be the major source of income, but you can't get them on board until you are set up "properly" for development and have attracted enough developers, and ideally paying developers at that...which is hard to do at present, maybe in part because you aren't showcasing the current sites?
What do you need in a showcase (initially at least) beyond a snapshot of the site, some blub, links to the site, to the project, maybe to the latest dev release?

What is the current business model - are any charities paying anything at all towards their projects/upkeep of the site?  Do you have an idea of how much you need to raise, essentially immediately, to stop you personally burning through money on this?  That has got to be somewhat of a priority right?   Perhaps once you get registered as a charity, you can do a wikipedia-like call for donations as a one-off, to ideally get AV funded until it can become self-sufficient?  It might be quite easy to get an iniital fund going to take the pressure off you...

## Premium Service ##
Yes, I think you need some sort of showcase when you start pitching to charities.
How did you find the existing ones?  Was it via personal contacts, either you or other members?

Why would a charity want to be listed on the AV site and what does it give them that the unlisted members don't get?  The primary way for people to find charities will be via google, not by going to AgileVentures, so isn't a listing just a "nice to have"?  
There are two things for the charities - one is being more visible to developers (perhaps this is what the listing is for?) so they get their site developed, and the other is to actually raise funds for the charity in question.

Perhaps one of the selling points to a charity to pay for a certain level of membership is that AV would run a fundraiser for that charity once they have been a member for a certain period of time?
This could be something IRL - though that might be hard to organise with developers being spread across the globe - or something in cyberspace - e.g a coding challenge with a small entry fee and a prize for the winner, with the remaining money going to said charity.

How many active developers do you have at the moment and how many genuinely active projects?
I'm wondering if you have enough to try to split developers into three tracks:
-- Working on existing charity sites
-- Improving AV for developers
-- Developing at least a simple showcase to attract charities?

Normally of course you would want developers to pick and choose what they work on but I'm sure in the early days people wouldn't mind being encourages towards one stream or another.
Perhaps the "discovering projects" screen would make it easy to find projects in dire need of some love, though I'm not sure what criteria you use to identify them - is a new project more in need of help than an existing one?  The ones with the most bugs?  The ones with the fewest developers?   The ones charities have paid to push up the ranks?

## Footnote##
Yes, it does have an "evolved" feel about it and it seems to be very heavily focused on the developers without focusing on making it easy for developers.
Perhaps you need to sweet talk a UX person into coming up with a good design?  The public face of AgileVentures is, after all, crucial for both Devs and Charities - you can still have AV devs actually implement the shiny new UX, but in my experience good developers !== good designers.
Indeed, I woud say it is so important that it could even come down to paying someone to design the site and then getting AVv devs to implement it...funds permitting of course :}

**Comment by freeranger** [comment]
A bit more thought, based on the slack chat.

## Up for Grabs ##
You seem quite taken with up-for-grabs.  If you want to move forward with it then I guess there are a couple of questions:
1. Are you so wedded to Pivotal that it is essential in the brave new world?
2. If the answer to 1. is No then would anyone be able to quantify the effort of moving away from Pivotal to UFG + github vs the effort of forking UFG and integrating it with Pivotal?
3. If the answer to 1. is yes then can Pivotal be modified to behave in a more UFG like manner? or would it make more sense to try to integrate the two?

I think I said before I see pivotal as more for structured teams running sprints (but perhaps thats what you want to achieve) than the more free form approach of the usual open source projects on github, so for me as a developer, I would be happy in the familiar environs of github.
OTOH, as a project sponsor, Pivotal will more easily let me track progress...so do we just say that devs have to suck it up and fit in with Pivotal?

If Pivotal stays, does it create confusion having two places to report issues - i.e. Pivotal itself and github issues tracker?   Perhaps if we can link between the two it wouldn't matter where the issue is reported, it would be visible in both?

## FreeCodeCamp ##
Regarding FreeCodeCamp, it seems quite slick.  It is clearly aimed at dev's learning, and in particular a certain set of skills (I would like to dub this the "Neeson" method in honour of Liam Neeson and his 'particular set of skills' :^D ), whereas AgileVentures isn't at all set up that way and doesn't have those goals - but one thing I think we can take from their site is the "gamefication" of learning - see how they use terms like "challenges"  and earning points for things like completing "challenges" or posting links, being upvoted, being thanked in chat etc.  

Similarly, Stack overflow do things like reputation points and badges for various tasks - leaving comments, providing accepted answers, even just completing your profile (http://stackoverflow.com/help/badges).

All this all encourages you (consciously or subconsciously) to keep going and try harder because you want more points - you want to feel like you are progressing and you want to let people see you have a bigger number than they have.  You can feel pleased with yourself and earn some bragging rights perhaps when you earn the Jedi Master Yoda badge for most accepted pull requests across all projects on the site, or your reputation points hit 10,000 or whatever.

AgileVentures could take some of these ideas to make the site more "fun" for developers and to encourage them to come back and contribute time and again.   How you monetise that is the tricky bit - are your badges only visible if you have made a donation or paid a subscription?  Or maybe you can only earn up to a certain level of points/badges as a free member and if you want to "unlock" the other badges then you need to pay up?

**Comment by tansaku** [comment]
Developers
--------

* *up for grabs* - maybe we don't need the whole thing - maybe we should switch to github issues? but then we can't assign people who don't have collaborator access I think and that maybe breaks the pull request model of a couple of the main projects - but each project is it's own master.  Every project team is completely independent and can decide their own workflow so it's up to them.  Either way it's probably a good idea for every project to submit to up-for-grabs just on the off chance; although it's a shame if we're not using all they have to offer;  which begs the question do we need a platform-neutral mechanism of sharing which tasks to start on, can we build that and how much of a priority is it

* **difficulty** Having multiple systems and repetition of issues is unDRY and to be avoided where possible.  What's the solution? Not sure .... there are lots of resources on the [relative merits of story points](https://www.google.co.uk/webhp?q=what%20are%20feature%20story%20points) and it's something that is introduced in the CS169 and AV102.  And if a new developer actually joins a standup and participates in an agile sprint they'll see the benefit of story points to the team and to communication and estimating progress etc.  Of course if they're opaque from outside, then that's a problem ..... as regards consistency between projects - we can't enforce it, we could have a template?  Is it desirable? I'm not sure ... part of the overall mission of agile ventures is to be sprint-oriented.  Github already provides almost everything people need for a "traditional" opensource free-for-all with random contributions any time, from anyone.  I think the point of what we are doing is to have something on top of that where you get the option to work more closely, in real time, with others as part of a more tightly knit team.

* **sample project** - yes something as you describe would be great to be built into AV102 or CS169 - would be great if somebody wanted to create something like this and we could add it to our documentation flow

* **guidance** indeed. Note that video capability is totally not required - although people keep on thinking it is for some reason and apologizing about it.  Personally I could not care less if you have video capability or not.  Is 99% irrelevant.  Fun sometimes if you have it.   What is critical is the ability to screen share, which everyone can do through google hangouts.   The not being able to make the time, and being shy/embarressed are perfectly good reasons however.  You say AV standups are at 5pm on a Fri?  That's just a test time for the weekly meeting.  There are standups 3 (or more) times a day every day at different times - see http://www.agileventures.org/about-us .... your points about emphasising their optional nature are well taken, and I guess what we need is to cohere around a vision and message and work to make the main site communicate that effectively.  Personally I am coming round to the idea that we should focus more on supporting people who want to work solo and asynchronously; although I'd like to see that as a gateway to ultimately more pairing and more attendance in standups (even text based standups see http://tatsu.io)

* **languages** agreed - would be good to make the options more visible
* **getting started** good point - and actually the key thing is just joining Github and AV.org to get into slack
* **search** glad you found it - point taken about tagging

Charities
---------

Well I'm not sure that charities will ever be a major source of income - they're not known for having lots of cash to throw around :-) Small charities are often trying to do things on a shoestring.  You're correct about what the showcase needs, not much.  I think the key thing it needs is to look good.  The problem with the group that built the previous one, and all credit to them since it was their first big project, was they got focused on things like login and authentication at the expense of CSS and UI/UX.

In principle if Thomas or I cleared a week and took a run at it we could have a simple showcase up that would be potentially very valuable in terms of attracting both charities and developers.  It's a question of priorities.

You ask about the current business model; we have had charities make some small contributions to projects.  I would need approx $50 a month to stop the sites current burn.  What I would like to set up is stripe integration so that people could sign up for an AV subscription of $1 a month or something to become a supporting member of AV.  Then if we get 50 people signed up we're at least not losing money.  I think that would be a great thing to start with and then iterate on the site and/or showcase to bring in more contributors and have premium subscriptions of $5 or $10 a month which gave more benefits in terms of promoting existing projects or supporting private projects.

Premium Service
---------------------

Existing ones have been found via personal outreach by myself and other members.  I also ran charity jamboree events at Makers Academy and streamed them live and got various charities involved by promoting that.

A charity might want to be listed to increase the chances that they'll get some low cost prototyping of solutions to their current challenges, and to be introduced to developers who'd like to help them.

Running fundraisers is an interesting idea.  Feels a little out of scope for me personally since that's what all the crowdfunding sites do and I'm not sure how we'd differentiate ourselves from them.  One way of looking at AV is that we are crowdfunding site where you donate engineering time instead of money.  

I think we have maybe 10 really "active" developers, and I'd say 4 really active projects although there's flux.  The three tracks you mention are there already implicitly - they could be emphasized ...

We totally need to sort out the project rankings and offer options to sort by "most in need of love" etc. :-)

Footnote
---------

Yes, a good UX person.  Although I'd say actually I'm a UX person - I've just been hands off on AV and been the client and not involved in the technical or UX side of it except as a "customer".  It was also about letting the learning developers on AV try out their own ideas.  Now I'm 100% on AV I think it's time to change my role and swing my weight behind really fixing the UX on the site and getting involved at all levels.  See http://harrowcn.org.uk as an example of where I've been more involved.  Not that we couldn't use all the help and input we can get.  I'm no UX guru :-)

Up For Grabs
-------------------

Is great partly because I know of various non-AV people who'd love to see something like that.  It's possible we might transition some projects away from Pivotal, but I feel it's important that we try to have a broad tent to allow different project styles to emerge.  The site currently gives projects a push towards github and pivotal, but I'd like us to support all combos, e.g. bitbucket and trello or whatever.

If we have the sample project you suggest, that can be a standard onboarding using github/waffle or whatever, but I my intuition is that we benefit more in the long run in terms of what we learn through diversity by allowing projects to pick and choose their approaches and tooling.

I'm in two minds about whether we want to try and make it trivially easy for a solo programmer to pick up a simple task asynchronously, OR trying to better communicate the benefits of getting involved in either synchronous standups, sychronous pairing, or both ...


FreeCodeCamp
---------------------
Is totally slick and I'm totally jealous :-) It is targeting complete noobs in a way we are not, but I think it benefits from a single coherent vision which we have not yet achieved, or at least not yet expressed through our site.

I don't think we want to target complete noobs.  Effectively CS169 is our onboarding platform - although maybe that needs to change.  But we should totally gameify.  I'm deeply inspired by codewars.com - I would like to see an explicit reputation score in our members list based on github contributions to non-profit projects ...

Great chatting with you about all this BTW - all my opinions are just those, and I'm totally open to changing my opinions :-)

**Comment by freeranger** [comment]
## Developers ##
**Free for all or curated development?**
I think you need to decide if you want every team to be autonomous and have their own way of working, or if you want to have some kind of common structure.  By including Pivotal in the getting started requirements, you are implicitly  trying to give them a structure which leans towards sprint based work, and you said that "part of the overall mission of agile ventures is to be sprint-oriented", which states it explicitly...so are they autonomous or not?  It seems they are partially autonomous :}

I guess in an ideal world, everyone would be able to work however they want, whenever they want, but
- some people need (or would welcome) guidance and/or structure
- you are already, in your getting started section, "forcing" people to set up a pivotal account, github account, google+ and youtube, which is a lot of requirements for people who are free to work however they want.
Given these requirements, you have to have a github account anyway to join, then why not make that THE platform for storing your source?  If people object to github for whatever reason then they are not even going to make it past the getting started section.

You want people to work in sprints, attend standup etc, so why not provide the complete infrastructure for working in this way - you HAVE to use githug, you HAVE to use pivotal etc?  If you don't like these then AV is not the place for you - go find a different sort of experience.

Obviously you don't really want developers to feel their choices are restricted, but at the same time, if you are providing an environment for them to hang out and some structures to work within then it makes it easier to move between projects because you re already familiar with the setup.  I don't really want to go from pure github on one to having to use pivotal and bitbucket or some other such thing I am not familiar with on the next project - it makes me less inclined to join that one because I have to learn too much - the project, maybe the technology, the way it is run, the platforms used...

Could you create a platform-neutral system that is so flexible you can plug in any issue tracker to any source control system? Yes.  Is it worth the effort?  Possibly not.

**DRY** - agreed, you don't really want multiple ways of doing things, (but then that goes against the idea of developers being free to choose how they work), so you either pick one existing solution and that is the one way to do things, warts and all, or your build your own that meets your exact requirements - this could be from scratch or could be modifying/creating bridges between existing components to work in the way you want.

**difficulty** You mention CS169/AV102 - so now is there an implicit "you will have taken one or more of these courses to get up to speed"?  Whilst you may get students from there moving to AV, surely you want the barrier to entry to be as low as possible for everyone, even people who have not taken these courses?  
Developers may indeed "get" the value of story points and sprints and standups if they join and participate...but you have to get them through the door first, and that may be difficult if there's too much "weirdness" and not enough "it looks like I can ease myself in gently to this"

**sample project** - again the reference to the courses - if AV is meant to be a progression from the courses then that's fair enough, but I didn't really think AV needed to be related to the courses, save that iCs169 forms part of the secret origin of AV.   Making it as easy as possible for anyone to join means, to me, that there is no link to the courses, no "prerequisite" to have done them or already be familiar with the setup and terminology used by AV - you want to help people to be better coders, so teach them what they need to know to get going with an AV project.

**guidance** - yes video does seem to be essential even though that is not the intent.
Leaving aside those reluctant to use video, it does also preclude those who can't even screen share due to bandwidth restrictions.  It's all very well in the US or the UK, or many other "first world" countries where you have great broadband to rely on, but some people are in a hut in a small village in africa sharing a poor 3G signal between 10 people - screen sharing would be impossible for people like that.
The tatsu.io idea is certainly an interesting one.  What would be great is if people can join into a standup in whatever way works best for them - so you are hosting a standup using a live video feed from your office but someone else joins in audio only over some lower bandwidth connection, and others (somehow) can join in on a text basis only.
_"we should focus more on supporting people who want to work solo and asynchronously; although I'd like to see that as a gateway to ultimately more pairing and more attendance in standups" - agreed, the first taste is free, get em hooked firs and then there's no getting away :)

**getting started** - so pare it back to the minimum - something like
1. Sign up to AV and github
2. Go create!

Then you can follow that with other, optional, steps, to get more out of the experience etc...

## Charities ##
Agreed that charities don't tend to be swimming in cash, but they could be a source of small amounts of steady income.  Yes they can hopefully get a site done for free, but if they are in a hurry or want some of the other benefits then thy pay a small sum of money towards that.

Whether they pay or not, you want to get them on board to give the developers something to do, so you might want to rethink this wording:
_"From an organisation's perspective it is best to think of collaborating with Agile Ventures developers for the purpose of better understanding the nature of the IT challenges faced.  In favourable circumstances this may lead to Agile Venture developers being able to help craft a solution, but this should not be an assumed outcome"_

I understand why you have to say something about not guaranteeing a site at the end of it, but this says to me "don't come here looking to have a site built, come here if you want to understand the IT challenges in building a site"...but I am a small charity and really I don't have the time to spend or the interest in learning about the challenges of creating a site - what I want is for some kind developers to donate some time to making me a website.

re: the showcase - whilst I can understand dogfooding - after all, if you don't trust your own developers to make your site, why would anyone else trust them - as I said before, there is a distinction between UX design and actual coding implementation of the design.  The AV developers can code the site for sure, so you are dogfooding to that extent but there's no reason the design can't be done by a "seasoned pro".
It also serves as a teaching exercise for AV developers to get an idea of "this is what a showcase site could look like", for those charities that want to showcase the work they do for example on their own (AV developed) site.

I'm still not sure whether the primary purpose of AV is to help developers to develop or to help good causes to get on the web for little or no money, but whichever is the primary purpose, that is where the focus should be methinks..

$50/month seems fairly easily attainable I should think, with $1/month subs from developers.  Jumping to $5 or $10/month needs a *lot* of added value - so much on the web is free that people are reluctant to pay for anything any more - In a world of free software, free hosting, free cloud drives, free music etc, if you are charging for something then it either has to be such a trivial amount of money that people don't mind paying, or it has to offer clear advantages over rival, free systems.
I'm sure you could get some people to kickstart the charity with some one-off donations to get a little bit of cash in the kitty.  Perhaps these even make them "founder members" or some other such thing.  Could be if you pay enough, your membership is free forevermore, with entitlement to all the premium benefits whatever they may be, now and in the future...just thinking of ways to inject a bit of up front cash...

## Premium Service ##
The fundraising idea was something to benefit both devs and charities - devs because the winner gets a cash prize (or a donation of premium membership?) and the charity gets some cash from the (low) entrance fee the devs pay to join the competition.  Even if they don't raise much, it raises the profile of the charity for a period of time.  
I wasn't thinking of a charity ball or anything like that :)
The charity jamboree events you mention are another avenue - premium charities get promoted at these things...regular charities don't.

_"One way of looking at AV is that we are crowdfunding site where you donate engineering time instead of money"_ - this harks back to one of my original comments, that developers are already contributing time to AV so to try to get them to stump up some cash for AV too might be a tough sell..

## Footnote ##
_You're_ a UX person?  You hide it well :P 
Yes, AV definitely needs some love and some thinking about who the primary "customers" are and what sort of "user journeys" they will be making through the site.  I mean sure, eye popping visuals are nice but the site needs to be functional first and foremost - easy to navigate, easy to find what you are after, easy to understand what AV is all about and how you can get involved.
If you get some UX guru involved then great, they can add in the scrolling marquee controls with purple text on yellow background etc... :)

## Up For Grabs ##
You can't focus on everything at once, so _maybe_ ultimate flexibility is laudable goal long term, but in the short term, K.I.S.S and focus on a good basic experience with one set of tools other than sort of supporting multiple platforms - if you can't do it well with one set of tools then you certainly won't be able to do it well with multiple ones!

_"I'm in two minds about whether we want to try and make it trivially easy for a solo programmer to pick up a simple task asynchronously, OR trying to better communicate the benefits of getting involved in either synchronous standups, sychronous pairing, or both ..."_
If you can make it trivially easy for a single developer to work asynchronously then they can be eased into the idea of synchronous pairing, but it is not true the other way around.  I repeat what I said above - you can't focus on everything at once - so if it is trivial for a solo programmer to contribute and it is also possible for them to pair up then great - you can shift focus to encouraging pairing and standups once you are happy that the solo experience works well.

## FreeCodeCamp ##
Correct, they target the total novice, where AV requires some level of knowledge.  
Whilst I can see benefit in encouraging CS169 students to join AV, I don't see the benefit (to AV) in having CS169 as your unspoken "on boarding" platform (I don't see it mentioned anywhere except in the secret origin section).
Tying to CS169 has several problems:
- People have to, (or feel they have to), take these courses or have equivalent ruby/rails knowledge, which they may not want to do 
- The projects will be basically all Ruby projects because that is what CS169 teaches, so your target developer audience is severely limited

Although it is not your intention to limit the skillset, tying to CS169 would effectively imply that.
It could be a suggestion, as could many other sites for learning your craft, but perhaps it gets no more merit than any other other route to learning programming skills.

The reputation thing needs careful thinking about because there are many ways to contribute to a project.
- If I push 10 small commits of bug fixes does that score more rep than 1 commit of a new feature?
- What about the documentation I updated?
- Or I'm a rubbish coder but a great organiser so I run all the sprints - do I get credit for that?
- I'm a good all rounder, but for this particular project I'm not coding, just testing the bejesus about of the site. I don't make any commits.  Do I not deserve a better reputation?
- I don't contribute code to any projects because I don't have the time to, but I contribute a lot in discussions on slack or gitter or other such forum for talking and helping people - surely I deserve recognition?   

I think this is where the stack overflow type thing comes into it - you get points for a variety of different activities you may have carried out that can be tracked.  The trick is to make it balanced and ensure that every type of contributor, as far as is possible, can get credit for their efforts

**Comment by freeranger** [comment]
## Premium Benefits - Developers ##
Developers love free stuff and they love great tooling, and people like Jetbrains offer free licences to open source projects (https://www.jetbrains.com/buy/opensource/), so can we bring the two together?

You normally have to apply to them yourself, have them review the project etc and it takes a bit of time and is a bit of hassle.
What if premium members, once they hit a certain level in terms or reputation or badges earned or whatever, automatically get licenses for these tools?
Obviously you would need to negotiate with jet brains and anyone else you wanted to get free licenses from, to agree that AV could curate the licences and if they say that Joe Developer meets the criteria because the project is Open source, active, he has made X commits etc, then he automatically is entitled to free licences for as long as he is an active, contributing member of the AV community.

**Comment by tansaku** [comment]
Wow, we're having a really long chat here @freeranger - thanks for taking the time to write all these thoughts out - I really appreciate it.

I'm not sure how accessible this will be to others, but great to have as a resource.

Developers
---------------

**Free for all or curated development**

You are totally correct that the current "getting started has too many requirements" - and I think the way to go is to reduce those requirements - I've just edited - please have a look:  http://www.agileventures.org/getting-started you can edit too - anyone logged into AV.org can.

People are autonomous.  We encourage in certain ways, but also waffle, trello and atlassian can be used for Sprint.  At the moment my intuition is that we shouldn't push for Pivotal for all projects.  I think all the great stuff the OSRA project did might not have happened at all if we had pushed too strongly for pivotal.

**difficulty** I am not sure we want the barrier to entry to be as low as possible.  That's what freecodecamp wants, but I think while we want to remove all unnecessary barriers, I think people joining AV do need a basic familiarity with git and github.  We can point people to resources to help, but I'm not sure there's such a big benefit to us scaffolding that when so many others are doing it.

However you are totally right that we have to get people through the door.  Maybe that's a reason to drop pivotal and waffle (which have story points built in) and move to something more flexible like trello or just github issues.  Story points are all over the industry so it's an important thing to learn ... perhaps we can communicate it better - maybe we need to phase introduction by starting with levels of 'easy', 'medium', 'difficult' to help newcomers, but it's all a balance.  We have limited resources - so where to focus them ...

**sample project**  I'm starting to think that LocalSupport is our sample project - you can't break it - I'm always on supporting it responding to any questions, pull requests, issues - I am not sure how much easier we can make it without making it trivial - maybe I'm wrong - I guess a practice-ground repo might make sense though - I'm conflicted :-)

I think that we should push people towards some pre-requisites, e.g. this will make more sense if you have a background in programming, have taken CS169, have completed a bootcamp ... maybe we need to make that more explicit.  We don't want to burn time educating people from scratch when that's what CS169/AV102, bootcamps and freecodecamp are already doing.

**guidance** we may be excluding some people on the basis of bandwidth, but if we drop that aspect of AV we are basically github, or a github skin, which is a different proposition, and bandwidth is fast increasing all round the world.  We need to occupy a niche effectively before expanding I think.

I would totally love to be technologically transparent for standups, so it was irrelevant whether you were participating via text or hangout.  Maybe we should make all our standups text, with an optional hangout at the end ... we'd probably get much higher participation ...

You're totally right that some times of day will not suit some people.  Absolutely - that's why we try to have multiple standups at different times to suit certain people.  The vision is that at critical mass you'd never have to wait more than 15 minutes for a standup because someone somewhere would always be running one.

But, yes, lack of ability to attend shouldn't preclude involvement, and it doesn't.  Just start talking in slack or on github issues and we reach out.  Of course that doesn't mean there isn't a lot we could do to make that clearer.

**getting started**

Yes, please help me edit http://www.agileventures.org/getting-started to pare it down :-)

Charities
-----------

Re that wording *"From an organisation's perspective it is best to think of collaborating with Agile Ventures developers for the purpose of better understanding the nature of the IT challenges faced. In favourable circumstances this may lead to Agile Venture developers being able to help craft a solution, but this should not be an assumed outcome"* maybe, but I'm tempted to be cautious there until we have more resources.  

Feel free to propose an alternate wording, but I think we need to be cautious in what we suggest to charities that they can get out of working with us.  I have some more people now chatting about the showcase, but I'm not going to make that my main focus until I have a payment system integrated into AV to receive donations to address our current burn.  Of course we can interate on our messaging again and again in conjunction with analytics with who is clicking through where in the site.

And yes, if we had funds, or the right volunteer - we could totally have a seasoned designer working with our developers on the showcase site.

Regarding the purpose, we have a twin purpose, help learning developers, AND help charities.

We've had one off donations before - that's a great start, but I think to level up we need a subscription model.

See https://www.pivotaltracker.com/story/show/91358028, https://www.pivotaltracker.com/story/show/67538948

Premium Service
-------------------

You say 

> "One way of looking at AV is that we are crowdfunding site where you donate engineering time instead of money" - this harks back to one of my original comments, that developers are already contributing time to AV so to try to get them to stump up some cash for AV too might be a tough sell.

Maybe, but I have people offering me money, and there are a lot of people who got their start in well paid programming jobs by what AV offered for free.  I think it's at least worth spending a couple of engineering days getting a subscription based payment setup in place to explore that.  However well or badly that works we iterate on what we offer premium members/charities until we hit on the right combination that stays within our twin goals of helping developer skill up and charities get their IT needs prototyped.

See https://www.pivotaltracker.com/story/show/108063300, 

Reputation
--------------

Agreed the reputation thing needs some thought, but not too much.  I think we should get something out and iterate.  We shouldn't agonise too much about making it perfect, which is the enemy of done.  10 small bug fixes and 1 feature are both great.  Documentation should be in github repo so that count's too.  

In the future starting standups, chatting in slack or testing the site is something we can pull in, but no need on the first iteration, let's get something out to encourage/reward contributions and see who starts complaining and then respond to users concerns.

See https://www.pivotaltracker.com/story/show/78088920

Premium Benefits - Developers
---------------

We have that deal with RubyMine - developers on certain projects get RubyMine licenses - perhaps that should become a premium feature as you suggest.

Summary
--------------

You've made lots of great points and I'm really enjoying the discussion.  I'm keen now to start following through on some of the points by making changes to the site, deploying things, getting feedback from our community.

I'm also off tomorrow, so apologies if I don't follow up as quickly over the next few days.  Either way going forward lets move to opening tickets in pivotal and making changes to the site.  Whatever time you can put in we will all be very grateful for.

I've put some links to existing pivotal tickets in the above - do search https://www.pivotaltracker.com/n/projects/982890 for more and perhaps we can break this discussion up into components and continue in the tracker?

**Comment by patmbolger** [comment]
Great discussion above.  One suggestion: create a high-level description of where we want to go and how to get there.  This can be as simple as: 1) list objectives to be achieved, and, 2) for each objective, define strategy(s) to be used to achieve the objective.

That might give us a little more clarity on where we can start (we could rank the objectives) as well as commonality among implementation efforts (a single "strategy" could support more than one objective).

I find this kind of thing to be a great starting point for more detailed planning, where there is a common context and understanding of the bigger picture.  

An example snippet, for a couple of objectives, might be (major bullet = objective, minor bullet = possible strategy to help achieve):

* Develop a robust model for expanding the AV talent pool
  - Expand AV associations / partnerships to include MOOC courses:
    * focused on other dev technologies
    * focused on UI/UX designers and Product Owners
    * focused on SW Project Management / Scrum Master
  - Structure some projects and/or cross-project roles as "paid" positions (assuming we develop a premium model for outside funding, and that this supports this concept)

* Develop a better project recruitment and on-boarding process
  - Define a project overview template that all teams would fill out - defining significant aspects of the project, including high-level description of project goals, technical architecture and technology choices (made or yet to be made), block diagram (or some other mechanism) depicting major chunks of functionality required, key project positions needing to be staffed., et.al.
  - Periodic project overview talks (communicated to the community well in advance)
  - Maintain a backlog of stories that a) could be handled by an inexperienced person (noob), and/or b) does not require deep knowledge of the application domain nor sophisticated system behavior.

There are probably a dozen or less key objectives to define, followed by discussion of how to get there.  Then, dependencies and synergies will be more apparent (e.g., a developing a talent pool and staffing a project could obviously be very tied in to revenue stream models).

**Comment by patmbolger** [comment]
Also, please note that the example ideas above are just that - I don't mean to imply that these specific examples are necessarily "real" at this point.

**Comment by tansaku** [comment]
great contribution @patmbolger !

**Comment by freeranger** [comment]
@tansaku - yes, I have been thinking this "chat" is getting a bit unwieldy so perhaps we need to draw a line under it and break it down into separate things for consideration.
Having said that, here's my response to your last post :)

## Developers##
**getting started**
Wow, any member can edit!  A bit dangerous no?  Ok you'd like to trust everyone but this is your main site and you don't want just anybody messing with it?  I have a grudge against you because I did badly in CS169, or even I am just a nasty hacker, so I register with AV and completely deface the site - ok, not something that is likely happen too often (hopefully!) but still a possibility.
Maybe only people working on WSO should be able to edit it?

Step 1 is definitely better, however the whole page still seems a bit intimidating - there are lots of steps and without reading them all, I don't know which ones are optional!
I think it needs to be a bit snappier - bare minimum and then all the other gubbins if you have more questions.  You want to demonstrate "it's *this* easy to get going, but if you want more info or to delve deeper, then do X, Y, and Z"
Question: Why is a github account essential?  What if bitbucket is my poison?
If I wanted to reword it completely, do I just go ahead and edit it?  Surely better to discuss (e.g. in a story?)
_Note: I'm not saying I want to completely reword it and ATM I don't have the time, but soon hopefully..._

**free for all vs curated**
I guess you need to strike a balance.  Personally I'm not mad on Pivotal and am used to JIRA in my work environment, (and of course many other options are available), but I'd prefer consistency over "my" preferred tool.
In some ways "use what you want" is good, but at the same time, might it discourage some people from joining a project? [groan]I need to sign up to _another_ service now?[/groan], nah I think I'll skip it.
A balance might be to strongly encourage one toolset, which might garner more support from AV, but allow you the flexibility of choosing others....but somehow still give a joined up experience, with minimal friction to getting involved in a project.
Of course, the trick then is to decide which toolset is the "preferred" one.
github is wildly popular of course, but bitbucket lets you have free private repos...you mentioned before a premium feature of AV might be private projects - but (for small projects) bitbucket gives you this anyway.  I don't know what model you had in mind - if github is the preferred source control medium then the only way to have private repo's is to pay for them - or did you mean privately managed (through AV?) but the source is still public on github?

**difficulty**
Correct, developers can't be starting from scratch - as in they have some programming (or other useful skills) but they may be novices when it comes to agile or story points or working on open source or git or all of the above - so I mean we want the barrier to entry to be as low as possible, so people with something to contribute can get stuck in as quickly as possible without having to spend ages setting up accounts all over the shop or scrabbling about trying to find out about the terms being used so they understand how to get involved.

Agreed that if you want to get involved you need to have some experience - whether this is coding or testing or documentation or whatever it is you want to contribute - you come here to hone your skills and/or to pick up more skills, but complete beginners who don't know one end of a computer from another need not apply :)

You can point people to things like CS169 or codeacademy etc (but perhaps not freecodecamp since they are total rivals!) - maybe even the individual projects could have a section suggesting where people might go to learn the skills to help out for that project.
There's no point sending someone to CS169 for a Visual COBOL based project for example :-)

Story points and other unfamiliar terms should at least get a link to something explaining them, and we do need to know what they mean:
Project Foo uses 1, 3, 5 as their scale, but Project Bar uses 1, 2, 3 - so if I am looking for medium difficulty stories then I need to know, per project, what the scale is they are using. 
TBH I do like the up-for-grabs thing of saying something is easy/hard etc - maybe if each project does uses it's own scale then this is something they need to configure on their project setup page, so our U-F-G like search can display this as "easy", "medium", "hard", to give a consistent meaning in the results set.

**sample project**
I still like the idea of a separate sample project that lets you go through the full lifecycle, wearing all the hats one would wear in a project, so you can see how easy it is to submit code, create pull requests, accept them, push the site out live etc.
I could be the best Objective Caml developer in the world but I have never worked with a source control system so I find this whole github lark a bit intimidating, and as for Pivotal, well at least I'd _heard_ of Github ;).  
Yes we can point them to some online training for these things individually but it would be good to have something end to end.
Are all sites deployed with heroku?  Again, I have no idea, I'm worried I will break something on a real project (or I want to start a project and I am worried it might be too difficult), so the sample project lets me see the whole lifecycle without worrying about working on a real project.

**standups**
Sure, bandwidth is improving all the time in many places, but there are still a huge number of areas where people have limited bandwidth and it would be a shame to have to exclude people simply because they have a slow connection, but I take you point about starting somewhere...
The other reason to allow text based is because for example I am at work or I am on a train.  I can probably get away with a text based standup at work and on train the signal is too choppy for video even if I wanted to standup in front of strangers as it were, so again, the text based option would allow participation.
Or perhaps people can participate asynchronously.
- Fred wants to participate but cannot make a realtime meeeitng.
- In good time before the standup, he sends a list of things he wants to discuss to the person who will host the standup (or perhaps there is a web page somewhere for this specific standup where the issues can be posted
- Wilma, hosting the standup, raises the issues with the rest of the team.
- After the standup, Fred has a chance to review the video and, in that same place where his issues were posted and the video is hosted, he can post his response, if any.

I know you would like a live standup somewhere in the world pretty much 24/7, so there is always _some_ standup that anyone can join, but that's bugger all use to me if I want to join Project Foo but their standup is at 3am my time.  Sure I could join the Project Bar standup at 8am over breakfast, but I don't have the skills or the interest in Project Bar, so that really hasn't helped me at all, and I have gotten the impression that I must participate in the standup or else I can't really help Project Foo...so that's me out of the game :/
Have you tried out tatsu.io to see how it actually works and see if there is something similar you could do with AV standups?
 
## Charities##
Yes I fully understand the need to be clear that you aren't guaranteeing anything.
Perhaps create a story to come up with alternative wording?
Twin purposes - I have had some more thoughts on that, but I will post them separately over the weekend (I need to get on with some work ;) )

Re: payments - are you thinking of having a donation button on the website?  Even if you aren't getting in streams of cash from premium developers, you may get one-off donations from them...

## Premium Service##
Hey, I was only repeating what you said about looking at it as people donating their time rather than money!
Of course there will be some people willing to donate time AND money and if the premium benefits are attractive enough, hopefully lots of them!

## Premium Benefits - Developers##
That's great that you already have an established deal with Jetbrains, but why not try to get it for the whole toolset, and yes as a premium service.
I'm a C# developer so Resharper is essential, but I also use WebStorm for (can you guess?) web projects, RubyMine for CS169, and they do other IDE's for C++, python etc.
If you could get them to agree that people get the *relevant* product licenses if they are working on certain projects then that would be great - ie no point having a RubyMine license if your project is in python for instance!

As I'm sure you know, developers working on non profit software can apply and get these anyway (if they meet the right conditions) so non premium developers are not excluded from the ability to get a RubyMine license, just that it is a lot easier for premium developers.
Oh and that RubyMine icon is out of date - it's version 8 now :)
I'm sure there are other developer tools out there where AV could broker similar deals for premium members..

## Reputation##
Sure you don't want to spend ages agonising over it, but I think you want to have a reasonable idea of the direction you want to go and the things you might want to include in it
 - is a bug fix worth 10 points or 1 point?
- How many more points does a 5 point story get than a 1 point story?  Does it depend on the project?  (In mine, I might not use story points at all, or 5 is a medium whereas in yours, 5 is a hard)
- what sort of scale are we looking at?  Does having 10,000 points mean something, like I live, breathe, and sleep AV or might I have that simply by posting a documentaiton update once a month for 3 months?  Maybe this doesn't matter?  
- Do documentation stories get the same number of points as development stories?
- Do we want to have badges?  If so, do they confer extra points, or is it just that you have achieved that level?  (You have submitted your first bug!  Your have had 10 posts accepted as answers to questions!  You have started your own project!)

Perhaps a starter for 10 points (pun intended :)) is something like getting points for each of:
- you have signed up to AV
- you have created a github account
- you have created any of the other accounts we suggest on the getting started site
- points for completing the sample project (once we have one!) exervice
- points for slack posts
- points for signing up to a project
- points for repo commits
- points for attending (or otherwise contributing to) standups
- a zillion points for getting a project into production

If there are badges then maybe things like:
- badge for first project signup, then 5, then 10?
- badge for first repo commit, then 5, then 10, 50, 100, 200
- badge for first slack post, then 50, 100, 200, 500, 1000....?
- badge for creating a project
- badge for getting first project actually into production

These are all off the top of my head - I haven't really looked at the stack overflow or other such merit systems to see what they are offering - we probably want to be similar, but different ;)

@patmbolger Thanks for contributing to the discussion!
Yes we definitely need to be looking at breaking this down into separate issues, either to become real stories or that need to be hammered out with further discussion to see if there's anything actually worth pursuing.

**Comment by freeranger** [comment]
## Developers! Developers! Developers!##
I've had a bit of a Steve Balmer moment ;)
I've been wondering if it is taking the wrong approach trying to focus on developers AND charities initially and the focus should just be on attracting developers.

It's going to be easier to attract developers than it is to attract charities, (particularly paying ones), so Agile Ventures initially is all about the developers and the rest, to a large extent, will hopefully "just follow".
The developers will come up with ideas for non profit projects and some of them may even have contacts in non profit organisations that are looking to have sites developed, so let the developers worry about the projects and the charitable causes and let Agile Ventures worry about the developers.

Agile Ventures actually sourcing work/bringing developers and charities together could follow down the line when there's enough good sites to showcase and enough developers to go around to actually work on new projects - at the moment there seems to be a shortage of developers anyway so there would be little point in attracting more projects if there is nobody to work on them.

Note: you would still need the showcase part of the site now, to show to potential developers to get them excited about the idea of joining and the sorts of things people are working on.  A handy byproduct of this showcase for developers is that it would also be useful in attracting customers who see the quality of work Agile Ventures developers can churn out.

For now the focus is on the developer experience, attracting developers - ideally paying developers at that - and making it a great experience for them.  So…how do we attract developers?  Cool projects to work on, cool technologies to use, ease of getting into a project and getting help and not feeling like a numpty, a cool place to hang out, show off, learn, teach.

I am a developer interested in working on an open source project, or even starting my own - what clear benefits are there to joining Agile Ventures over and above free GitHub (even we are using GitHub!) or up-for-grabs or one of these other sites where developers can work on open source projects?  What's the USP?  This becomes even more important when trying to monetise the developers and I think it's gonna be hard to stand out from the crowd.

Here is one idea for premium developers: Smart Search
Many of the other sites have an easy way for me to search for projects I am interested in, whether it is just a general search or keyword/tag search - but what happens when I can't find any interesting/relevant projects today? Do I try again tomorrow?  The next day? Next week?
What if I could key in criteria for projects I am interested in - so I want anything which involves ruby AND angularjs OR is something to do with sloths.  I am only interested in brand new projects, or maybe I am only interested in projects started by people with at least 10,000 reputation points.
Every day when I log into AV, (or via email notification?) I will be informed of any projects which meet the criteria I have set up, and I could have multiple sets of searches, each with multiple matches.  The matching algorithm would rank the matches and show me the most relevant ones first.

Ok, that bit is not much different than the usual sort of search you might get on any number of rival sites where you are looking for projects, but what about the other side of the coin?
I have an idea for a project and I want to find people with the necessary skills who I can try to tempt to work on my project.
I am looking for people with C#, jQuery, and Mongo skills, I am only interested in people with at least 1,000 experience points or who have contributed to at least 2 projects or who are based in my time zone to make it easier for stand ups etc.  I am also willing to act as a mentor to one person who wants to learn the project management side of things.
I am particularly interested in people who are multi lingual who have worked on realtime interactive sites.
At project inception stage you may not know what technologies you want but you can still search for people who are looking to work on projects without specifying coding skills.
Again, the matching algorithms will show me the best matches for my criteria, and I will be notified when any new matches appear, and be easily able to contact these people about my project.

You could even take it further and say how many people with each type of skill you are looking for (2 c# developers with 3+ years experience, one mongo dude, as many JavaScript devs as possible) and you continue to get notifications of matches (you haven't already rejected) until your "quota" is filled.

This would obviously need people to upload their skill set to their profile and be open to being contacted by interested parties, which is either by opting in to be searchable or by specifying your own search criteria there you are touting for work.

Non premium members could look at any member and see their particular set of skills, or see that they are interested in doing etc, but they don’t get the auto updating power search capabilities.

Another idea for members, not necessarily premium (after all you probably expect to have to attract non paying devs first and then entice them onto the paying model once they fall in love with AV - but you have to get them there first) is mentoring - get people to agree to mentor to new members for a short period of time to get going on their first project. They would, of course, earn a badge or some such thing for doing this.

**Comment by tansaku** [comment]
@freeranger thanks for the ongoing input - haven't had a chance to process it yet - hope to look at it in detail soon - do check out the 2nd weekly meeting we had if you haven't already: 

https://youtu.be/0YQNKccX-Tw

https://docs.google.com/document/d/1qKLXYf5oTuT-A2eWVlh8_TBVfQHAwB8qr3HmzkOSEcQ/edit

Also we're looking for other times that you might be able to make - care to suggest some?

Also kick off meeting for WSO on Wednesday will be where we work out specific actions based on the high level direction:

http://www.agileventures.org/events/websiteone-agileventures-home-site-weekly-kickoff

**Comment by freeranger** [comment]
@tansaku Wow, 2hrs 43 - I'd best get my popcorn in ;)

I've watched about 15 minutes of it so far - will get back to more later.
From what I have seen so far and the google doc I have read, a few things stand out at the moment:
- having the time to devote/time management
- on boarding new developers to a project, keeping them, and getting them contributing
- Need for Project managers

## Time ##
For the former, I think this in part comes back to what seems to be a requirement to pair and to attend meetings.   Not only do I need to find the time to devote to AV projects but it also has to be time that will work for others too.  This may not be possible - I have a 1/2 hour or hour here and there on an adjective basis but since I haven't arranged a pairing session then what... I can't contribute? 
I don't want to or I can't commit to a specific time each day or each week to devote to AV projects - it feels too much like a job then and I (hopefully) already have one of those!

I think it needs to be made abundantly clear that people can contribute what they can, when they can - it is _better_ if they can pair and if they can attend meetings, and you will probably get more out of it that way, but by no means is it essential.

Further down the line, (way down the line), perhaps someone could put together some sort of tips or links to articles or whatever whatever on managing your time, working in a distributed team etc.
AV102 has a lot of good material that is not related to being a TA and not related to supporting CS169 so perhaps some of that, or variations of same, could be made available?
Like how do I set up a hangout?  do github pong etc?

## Onboarding ##
The up-for-grabs type of tagging mechanism, possibly combined with a mentor in the early stages I think would go a long way to address this.
If I want to get involved in any project, then there are some key questions that need answering:
- what is it about? Am I interested in this area?
- what technologies does it use?
- is it live?
- how do I get it up and running?
- does it have any issues I can work on?  (distinct from any issues that someone familiar with the project could work on)

I don't think it is easy to answer these questions with the current AV site and, picking the OSRA project totally at random, this is what I find:

**What is it about?**
The top of the project page says 
>OSRA is a non-profit organization registered in UK, aiming to sponsor orphans in Syria & ensure basic life & education requirements for Syrian children.

And then the About tab says:
> This project's objective is to build a support system for administration of the information needed to fulfill the organization's goals.

Ok good, short and sweet.  But more info might be nice but it's a start, though so far, unless the orphan aspect is what appeals to me, there is nothing to "excite" me about this project - why would I pick it over any others?

**What technologies does it use?**
I have no idea - there is nothing on the main project page about it, or on the about tab....going into the "Documents" section I see there are some things like a "getting started" guide and some technical discussion documents...maybe one or more of these will illuminate me on the technologies used by this project.   I know (SPOILER ALERT!) that it uses PostgreSQL because the getting started guide states that explicitly, assuming I haven't already moved on to the next project by now, and maybe some of the technical discussion documents will elicit some more info, but really, I have no idea! 

**Is it live?**
Well it does say "Active" but I don't know what that means - Refuee Tech was created about a week ago and that is "Active" but they haven't started coding yet!  So "Active" doesn't mean "Live".
There are no links to any actual live systems or even a running dev instance...maybe this project is actually not active anymore, just the site is out of date...

**How do I get up and running?**
Luckily, there is a "Getting started with OSRA" guide which I found on the "Documents" tab - why isn't this more prominent?  I have to dig around for it!
Ok, here we go....

>A project member will be glad to help you get set up in a live session - please email ... or post a request on our Slack channel if you already have access there. However, if you prefer, you can carry out the following setup steps on your own:

email?  Who do I email?  Slack channel?  How do I find it?

>Please read and follow Agile Ventures' general Getting Started guide. You will need accounts on Agile Ventures, GitHub, Google+, Pivotal Tracker and Slack.

wow, so many things just to get going - I just want to have a quick scoot around to see if I am interested or capable of contributing to this.

>OSRA uses a PostgreSQL database. The easiest way to get this working on a Mac is by installing postgres.app.

Finally, a concrete piece of info about one piece of technology the project is using!  Unfortunately I am running on windows or linux or some other weird and wonderful platform so the Mac advice is not much use at this point (_Note - I do recognise that telling people how to get the underlying tools running is likely beyond the scope of what AV would want to offer as it is outside of any project specific requirements_)

>Visit our GitHub repository and fork it to your own GitHub account.
Clone the OSRA repository to your computer with git clone https://github.com/YourName/osra.gitor git clone git@github.com:YourName/osra.git depending on your GitHub configuration.
Add the Agile Ventures OSRA repository as 'upstream' remote with
git remote add upstream https://github.com/AgileVentures/osra.git

Cool yes, github, I know about that, I can do that -lovely.

>Run bundle install
Run bundle exec rake db:setup and bundle exec rake db:test:prepare
To ensure that the setup worked, run bundle exec rspec spec/ and bundle exec cucumber and make sure that all test pass.

Wait, whatnow?  bundle?  rake?  what are these?  remember I still have no idea what this system is written in!  None of these commands are familiar to me, I'm off!  Luckily I read ahead to this point and didn't waste time installing PostgreSQL or setting up github before I realised this is not a project I can contribute to.

>The branch of the repository you forked and cloned is called 'develop'. You should do all work locally in a new feature branch, then push that branch up to your GitHub repo and submit a pull request for your code to be merged into the official OSRA 'develop' branch. 

Ok, if I am a lone developer then this all makes sense to me, but AV is all about pairing so I'm a bit stumped as to how I would work locally but also pair at the same time...with someone else with their own local copy?  Not totally off-putting but it is a question mark nonetheless....

Also, I note that someone logged an issue on this page some four months ago
_"I cloned the repo but got FATAL: Peer authentication failed for user "postgres" when I run bundle exec rake db:setup. The postgres is started. What did I do wrong?"_
but nobody has responded to them....so maybe this project is dead after all...or maybe the other members just don't bother helping n00bs...I'm off again!

Alight so I've made it this far - I have an inkling of an idea what the project is about, I know the database technology the project uses, (though I don't know about any other technology it uses) and I now have a clone of the repo.  There are some unfamiliar commands I am supposed to execute to check some tests run, and I have no idea how to actually run the application (website?  cordova app?  native Android app?) but lets move on...

**does it have any issues I can work on?**
Well there is obviously a github repo and github does have an issues section, so I go back to the repo and find that there are two issues from 2014 - maybe the project is dead? or just perfect as is and doesn't need any work?

Don't give up....back to the AV site and there is an "Activity" tab.  I go into that and coincidentally, there are two issues are listed..though these aren't the same two issues as in github, and they too are from 2014.

I notice under the "github" link on the right hand side of the project page _"OSRA - Support system on PivotalTracker"_ so I click into that.  I am not familiar with PT but I see the same two stories as on the "activity" tab, plus a bunch of others, but they are from 2014 and  there isn't much detail about them and they assume I know stuff about the project already.

Finally, I recall there was a bit more text on the "about" tab, so I go back there in one last ditch attempt to get going

> Note: the team is now using JIRA as the project management solution. Our scrum board can be found here. Our Pivotal Tracker board is no longer maintained.

Ok, it would be nice if the PT links weren't there and the JIRA link was more obvious but okay lets get cracking....I click on the JIRA link and get a login screen....what the?  Nobody told me I needed to create a JIRA account in addition to all the others I had to create.

The page says _"To request an account, please contact your site administrators."_, but I have no idea who the site administrators are or how to contact them - is it the admins for the AV site? OSRA? JIRA?  Sorry, life's too short, I'm off to up-for-grabs to try my luck there!

Now I know it seems like I'm picking on OSRA but I'm not - randomly clicking around AV, this has more documentation than many, and at least tells you how to get going with the repo!

Snow Angels, again picked at random for example has nothing about the technology, no getting started guide, seems to use Trello (another new tool! but also has a Pivotal link and, importantly, it has a link to a prototype site!  Score one for Snow Angels!

This sort of thing make it hard to become engaged in a project in the first place unless I am really determined, never mind the ongoing issues of having free time, being able to pair, getting up to speed with the code base, with the problem domain etc etc.

## Project Managers ##
There seems to be some discussion about requiring these for a project - I think that would discourage me from starting a project that it's suddenly this whole managed affair - I'm not into that, so I can't start one?  Or I need to find a project manager and convince them to join my project?

On the flip side, joining a project, I might like having a project manager running things - they will be a point of contact I can turn to if I have any issues (particularly early on) but also I want to concentrate on the development side of things so I have no interest in running sprints and meetings and whatnot.

I do think the idea of having a backup PM or pairing is a good one, where you choose to go the PM route - but then I would say that about any of the roles, it is good to have more than one person on it, be it testing, documentation, coding, scrum master, gofer, whatever...
It can serve as an opportunity for people to learn or hone their craft, or get into an unfamiliar area and have the benefit of working with someone experienced.

This sort of thing will come down to how much of a curated experience you want to offer developers - is it a free for all anything goes, or is there a framework that people can or even **must** work within, otherwise there's no point in them being here?

## Time##

In terms of making meetings, week days I can never make audio or visual meetings during the day - though I may be able to join a cheeky text based one or two.  
Evenings, I am generally travelling from 5 till about 6:30pm and could participate  by text for only part of that journey (not the driving bit for example ;) ).  
I am juggling things around a bit at the moment but currently it looks like Mondays or Thursdays would be do-able for me from about 6:30pm-7pm for 1/2 an hour or so.
Wednesdays are basically out until I get home around 8:30pm

Weekends are better - 6:30am-8:00am Saturday or Sunday and then from about 10 am onwards on either day I can generally make some time available.

Currently doing 2 x Coursera specialisations + the 169 TA'ing so actually getting stuck in to actual real work will likely have to wait a few weeks - one specialisation finishes about the same time as CS169.  
I can try to fit in things like reviewing documents or writing things up in the meantime.


Sorry that post ended up really rather longer than I intended but hopefully it proves useful to you.

**Comment by freeranger** [comment]
## Meetings ##
Standing up, squashed, on the train on the way home, I decided to listen to some more of the meeting.
No point really trying yo watch it even if I could raise my arms because it's a bit hard to read on a phone screen 😅
A couple of things struck me:
- How much mobile bandwidth am I burning "watching" this video I can't see?
- listening to the audio gives me the vast majority of the content and, if I cannot be present for a video meeting, this is is a great way to catch up - walking to the train, in the car, standing squashed on the train.  I have a lot more opportunities to listen and as a byproduct, don't eat into my data allowance so much ;)
- if it was anyone other than Sam talking, if it was an audio track,I probably speed it up to 1.25 or 1.5 times and still understand it perfectly - Sam, you already sound like you are speeded up a bit ;)

Do you have a feel for how often these standups and other meetings are viewed afterwards by people who maybe wanted to attend but couldn't? i wonder if audio versions might increase the audience?
So then this led me to wonder
- could there be a live audio only feed for meetings?
- could there be a non live audio stream after the fact?
- could these be turned into podcasts?

The last might sound a bit mad but even if you aren't involved in any projects, it may be interesting to listen to, to understand the problems of running agile projects, running remote teams, or just to get a feel for what else is going on in the AV world.  A podcast could be automatically downloaded to my phone overnight and I listen on the way to work...sweet.

## Search##
You were talking about me on the (I want to say podcast!) and my discussion on search.  I agree this was quite low level, but at the same time it was pertinent it to the 50,000 ft view - ie, what is the main purpose of Agile Ventures?  If you want to attract developers then it is essential that the site is accessible to developers, to find projects they are interested in.
If you want to attract clients first and foremost the. It is essential that the site showcases the work you can do.

## Charity##
I had a brief look at the charter and it is not exactly light bedtime reading! I know it is a template and necessarily will be couched in jargon, but perhaps it is worth pulling out the key bit, in English, about what the aims are of AV, so that you can get comment on it - I guess this is really the bit you want comment on because most of the rest will be necessary jargon.
I wasn't aware of the term "third sector", which could just be showing my ignorance, or could be showing your assumptions about people's knowledge since you already have knowledge in that area - in the interests of accessibility, perhaps keep the jargon to a minimum in the bits you want comment on - or at least provide a glossary of terms!
Also, as soon as you can, get a big old donate button going so you're not just burning through your own money while trying to figure out where AV s going and start the process to get there - you can start to burn other people's money too ;)

## Focus##
There was some talk about scaling up (# of developers) before scaling out (# of charities) but then that changed to getting our existing house in order and improving the current experience before we start scaling in any direction, and that may have changed again in the remaining two hours I haven't listened to yet!  There was also mention of lack of focus/direction within individual projects, so perhaps this is something to focus on (pun intended!) improving?

**Comment by tansaku** [comment]
We need to review this and extract more stories - @freeranger has made a good start with #763, #762 and #761
